libs/payments/cart/src/lib/cart.error.ts

```diff
@@ -82,3 +82,39 @@ export class CartInvalidStateForActionError extends CartError {
     });
   }
 }
+
+export class CartTotalMismatchError extends CartError {
+  constructor(
+    cartId: string,
+    cartAmount: number,
+    invoiceAmount: number,
+    cause?: Error
+  ) {
+    super('Cart total mismatch', {
+      info: {
+        cartId,
+        cartAmount,
+        invoiceAmount,
+      },
+      cause,
+    });
```

Comment:
This doesn't look quite right. The 2nd parameter of the CartError constructor takes an object that is passed on to BaseError as `info`. No need to wrap it in `info` in `CartTotalMismatchError`.

```
constructor(message: string, info: Record<string, any>, cause?: Error)
```

Suggested change as follows
```
super(
  'Cart total mismatch', 
  {
      cartId,
      cartAmount,
      invoiceAmount,
  },
  cause, // I'd recommend removing this. See previous comment.
);
```